homeassistant/components/frontend/__init__.py

```diff
@@ -44,11 +51,28 @@
 
 DATA_PANELS = 'frontend_panels'
 DATA_INDEX_VIEW = 'frontend_index_view'
+DATA_THEMES = 'frontend_themes'
+DATA_CURRENT_THEME = 'frontend_current_theme'
+DEFAULT_THEME = 'default'
 
 # To keep track we don't register a component twice (gives a warning)
 _REGISTERED_COMPONENTS = set()
 _LOGGER = logging.getLogger(__name__)
 
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Optional(ATTR_THEMES): vol.Schema({
+            cv.string: OrderedDict
```

Comment:
I think we shouldn't whitelist, so it is not tied to what the main backend can support.